src/corona_hakab_model/manager.py

```diff
@@ -159,13 +159,21 @@ def setup_sick(self):
         """
         agents_to_infect = []
         agent_index = 0
+        agent_permutation = list(range(len(self.agents)))
+
+        if self.run_args.randomize:
+            self.logger.info("creating permutation")
+            shuffle(agent_permutation) #this is somewhat expensive for large sets, but imho it's worth it.
```

Comment:
For large sets you can split them to smaller sets and shuffle each one separately. For this you should look at what size is performance an issue. Lets say we have 100,000 agents, then if we shuffle them in buckets of 10,000 I think we should be okay.

Reply:
We could, but shuffling 10,000,000 agents took about 10 seconds on my laptop, so I think that would be a premature optimization.